### ISSUE TITLE
Request: (ASP Net Core 5), option to Hide 404 responses

### ISSUE THREAD

**Issue by LordMike** [MiniProfiler/dotnet#570]
I have `.UseMiniprofiler()` set _after_ my `.UseStaticFiles()` - and before `.UseEndpoints()`, as I think is recommended. This means that Miniprofiler does not see any static files ... unless they're 404'ed.

When developing, my browser makes requests for `.map` files for js content, which doesn't exist with me - and in production, other files are 404'd that random browsers may ask for.

All those requests end up in the MP list, as they pass the static files middleware and so on.. I couldn't find an official way to hide it, but I made this middleware:

```csharp
app.UseMiniProfiler();
app.Use(async (context, next) =>
{
    // Hide all 404's from Miniprofiler
    await next();

    if (context.Response.StatusCode == 404)
        await MiniProfiler.Current.StopAsync(true);
});
```

So perhaps MP could be extended to:

* Have a "ShouldDiscard", like ShouldProfile, `func` that is called before `.Save()` which returns a boolean.. 
* .. or have a list of status codes or others, that should discard profilings.

I can do a PR for either, if it has any interest... I much prefer the first, as it will allow the user to do any logic they could think of.

Mike

**Comment by NickCraver** [comment]
Hey there! I'm glad you were able to do what you want here, but it's not something likely added to MiniProfiler for 2 reasons:
1. It'd be a breaking change to adjust the interface, both binary and source compatibility-speaking.
2. What would the "ShouldDiscard" method take?

For #2, given you may want to stop a profile for any reason at any time based on any parameters, making this generic enough would be hard. It's best to eject and stop profiling ASAP, so even the call location isn't ideal in a central call site if we did have all the desired parameters.

I hope this makes sense and is agreeable - the intent is definitely to allow cancelling from _anywhere_ as you're doing, and I am glad you were able to make that work well :)

**Comment by LordMike** [comment]
I imagine that adding a property to `MiniProfilerOptions` like this:

```
public Func<HttpRequest, bool> ShouldDiscard { get; set; } = _ => false;
```

I've simply copied the definition for `ShouldProfile` - maybe more info should be given, like a context of sorts with profiling data. On the other hand, the `MiniProfiler.Current` is always available for inspection, like in my code.

For compatibility.. I see.. it's not sealed, so the only breaking change would be binary for those who derive from the `MiniProfilerOptions` class.... that sucks :)..

I'll keep going with our solution.. I did get a null-reference exception during debugging once, where MiniProfiler.Current was null, so maybe there are drawbacks :).

Thanks for replying.
